### ISSUE TITLE
Fehler mit Matches Module in 3.0 Beta

### ISSUE THREAD

**Issue by cbabione** [Contentify/Contentify#596]
Hallo

Erstmal, danke für geiles CMS!!!!  Weil ich tägliches Deutch kann, möchte ich dieser Fehler auf Englisch erklären.

The Matches module for 3.0 Beta throws an error when called from the main home screen.
The error given is:
"Type error: Return value of App\Modules\Matches\Match::scopeFilter() must be an instance of Illuminate\Database\Eloquent\Builder, none returned"

The function on line 145 of /code/app/Modules/Matches/Match.php is set to typecast the return as Builder.  A Builder object will not be returned from the function if there is not a team_id specified.  
Current Broken Code:
public function scopeFilter(Builder $query) : Builder
    {
        if (ContentFilter::has('team_id')) {
            $id = (int) ContentFilter::get('team_id');
            return $query->whereLeftTeamId($id);
        }
    }

Removing the Builder typecast for the function allows the main home screen to work as expected.
Working Code
public function scopeFilter(Builder $query)
    {
        if (ContentFilter::has('team_id')) {
            $id = (int) ContentFilter::get('team_id');
            return $query->whereLeftTeamId($id);
        }
    }

I am not sure of your coding standards for this project, so will let you decide the best fix, perhaps either removing the typecast, or adding a return if the query does not contain a  team_id

All the best, and Awesome work so far!!!

**Comment by cbabione** [comment]
This also works by just returning the query:
    public function scopeFilter(Builder $query) : Builder
    {
        if (ContentFilter::has('team_id')) {
            $id = (int) ContentFilter::get('team_id');
            return $query->whereLeftTeamId($id);
        }
        return $query;
    }

**Comment by chriskonnertz** [comment]
Hello,

ugh, I am not sure if PHP 7.0 supported nullable  return types... try `?Builder` at the end of the first line.
In case that doesn’t work try to remove `: Builder`.

**Comment by cbabione** [comment]
remove : Builder works and was in the original comment.  Also, just returning the $query works too.  I just wanted to share my findings to help move 3.0 out of beta!!  Awesome work guys.

**Comment by chriskonnertz** [comment]
Soooo... do I have to change the code? 🤔

**Comment by cbabione** [comment]
I just created a fork for your 3.0-dev branch and will get a pull request to you for review.  Yes, code does need to change, but let me finish up in my fork and you can see what I changed.  Also, I found another bug with the User model and will include that too.

**Comment by cbabione** [comment]
I now understand the problem.  Originally I downloaded the 3.0 beta zip file and it had the problem I described.  Now that I have forked, the Matches Module in your current code contains the fix I described in my second comment, returning the original query.  Also, the error I saw with the User model is not in the latest code either of your 3.0-dev branch.  My apologies for opening this ticket as the bugs described were in the zip file downloaded, but have since been fixed in the 3.0-dev branch.  There does not seem to be a need for any code changes with this issue as it looks like it has already been done.
